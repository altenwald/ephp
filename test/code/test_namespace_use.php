<?php

namespace My\Full;

class Classname {
    function __construct() {
        print "Namespace => " . __NAMESPACE__ . "\n";
        print "Class => " . __CLASS__ . "\n";
        print "Method => " . __METHOD__ . "\n";
    }
}

namespace My\Full\NSname\subns;

function func() {
    print "running function " . __FUNCTION__ . " from " . __NAMESPACE__ . "\n";
}

namespace Code;

use My\Full\Classname as Another, My\Full\NSname;

print "Running from: " . __NAMESPACE__ . "\n";

$obj = new Another; // instantiates object of class My\Full\Classname
NSname\subns\func(); // calls function My\Full\NSname\subns\func
