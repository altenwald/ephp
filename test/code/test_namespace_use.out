Running from: Code
Namespace => My\Full
Class => My\Full\Classname
Method => My\Full\Classname::__construct
running function My\Full\NSname\subns\func from My\Full\NSname\subns
